priam/src/main/java/com/netflix/priam/tuner/PropertiesFileTuner.java

```diff
@@ -1,87 +1,55 @@
 package com.netflix.priam.tuner;
 
-import java.io.File;
-import java.io.FileReader;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-
-
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Splitter;
+import com.google.inject.Inject;
+import com.netflix.priam.config.IConfiguration;
 import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.configuration2.ex.ConfigurationException;
-import org.apache.commons.lang3.text.StrSubstitutor;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.text.StringSubstitutor;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.inject.Inject;
-import com.netflix.priam.config.IConfiguration;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
 
 /**
  * Support tuning standard .properties files
  * <p>
  */
 public class PropertiesFileTuner
 {
-    private static final Logger logger = LoggerFactory.getLogger(JVMOptionsTuner.class);
+    private static final Logger logger = LoggerFactory.getLogger(PropertiesFileTuner.class);
     protected final IConfiguration config;
-    protected final String propertyPrefix;
 
     @Inject
-    public PropertiesFileTuner(IConfiguration config, String prefix) {
+    public PropertiesFileTuner(IConfiguration config)
+    {
         this.config = config;
-        this.propertyPrefix = prefix;
     }
 
     @SuppressWarnings("unchecked")
-    public void updateAndSaveProperties(String configPath) throws IOException, ConfigurationException
+    public void updateAndSaveProperties(String propertyFile) throws IOException, ConfigurationException
     {
-        File propertiesFile = new File(configPath);
         try {
-            if (propertiesFile.exists() && !propertiesFile.canWrite()) {
-                throw new IOException("Can't write and therefore cannot tune" + configPath);
-            }
-
             PropertiesConfiguration properties = new PropertiesConfiguration();
-            properties.getLayout().load(properties, new FileReader(propertiesFile.getPath()));
-
-            Set<String> keys = new HashSet<>();
-            properties.getKeys().forEachRemaining(keys::add);
-
-            Map<String, String> overridenProperties = new HashMap<>();
-            String propertyOverrides = config.getProperty("propertyOverrides." + propertyPrefix, null);
-
-            if (propertyOverrides != null) {
+            properties.getLayout().load(properties, new FileReader(propertyFile));
+            String overrides = config.getProperty("propertyOverrides." + FilenameUtils.getBaseName(propertyFile), null);
+            if (overrides != null) {
```

Comment:
Would this code handle overrides being an empty string?

Reply:
Good catch. It throws an IllegalArgumentException which I have confirmed in practice. Moreover, it will throw when overrides resolves to a map with missing or duplicate keys or missing values as well. This will cause Priam to fail to tune the yaml and report unhealthy. 

I assert this is the correct behavior as Priam will continue to start, thereby minimizing the potential impact on other scheduled jobs like backups, but also doesn't just fail silently as was the case previously.

Reply:
That makes sense, is it possible to implement a default behavior if overrides is an empty string?

Reply:
What default behavior do you think is best? 

My preference would be to ensure the following hold:

1) different user inputs result in different behavior
2) overrides overwrite specific properties in the file as specified in the String 

Assuming we adhere to these precepts, I think throwing on empty string is best because it prevents users from mistakenly thinking they've overwritten something when they haven't.